client/lib/json-patch-lib.ts

```diff
@@ -15,74 +15,68 @@ export interface AddOperation<T> extends BaseOperation {
 
 export interface RemoveOperation extends BaseOperation {
   op: PatchOp.REMOVE
+  value?: undefined
 }
 
 export interface ReplaceOperation<T> extends BaseOperation {
   op: PatchOp.REPLACE
   value: T
 }
 
-export type Doc = { [key: string]: any }
-
 export type Operation = AddOperation<any> | RemoveOperation | ReplaceOperation<any>
 
-export function getValueByPointer (doc: any, pointer: string): any {
+export function getValueByPointer (doc: Record<string, any>, pointer: string): any {
   if (pointer === '/') return doc
-  const pathArr = pointer.split('/')
-  pathArr.shift()
+
   try {
-    return pathArr.reduce((acc, next) => acc[next], doc)
+    return pointer.split('/').slice(1).reduce((acc, next) => acc[next], doc)
```

Comment:
```suggestion
    return jsonPathToKeyArr(pointer).reduce((acc, next) => acc[next], doc)
```
where `jsonPathToKeyArr` is a more explicitly named function of the operation, so we know what it's doing and why in the future. If you're feeling up for it, adding a small testing framework and case for this function would be useful.